Priority Queue/FastPriorityQueue.cs

```diff
@@ -365,7 +407,8 @@ public void Remove(T node)
             #endif
 
             //If the node is already the last node, we can remove it immediately
-            if(node.QueueIndex == _numNodes)
+            int index = node.QueueIndex;
```

Comment:
No change even with the repeated uses of the value 10 lines further down? As I understand it local variables are faster to use than properties, and this particular value is used three times. Compiler might be smart enough to optimize that, though. I'm fairly sure I tested the effect of this specific change and saw some small but consistent improvement, but I didn't keep my records of intermediate test times.